Add Mastra Integration Documentation for Arcade

akuya-ekorot opened the pull request:
<!--
Before submitting a pull request, please read:

For contributing a community toolkit, please use the [Community Contributed Toolkit Pull Request Template](https://github.com/ArcadeAI/docs/blob/main/.github/PULL_REQUEST_TEMPLATE/community_contributed_toolkit.md).
-->

## Overview
This PR adds documentation for integrating [Mastra](https://mastra.ai) with Arcade, enabling developers to use Arcade's extensive tool ecosystem with Mastra's agent framework. The documentation explains how to convert Arcade tools to Mastra-compatible format and handle user-specific authentication flows.

## Changes
- Added step-by-step guide for integrating Arcade tools with Mastra agents
- Implemented JSON Schema to Zod conversion explanation with appropriate callouts
- Added examples for Next.js applications
- Included dynamic toolsets pattern for per-user authentication

## Related Documentation
This complements existing documentation for other frameworks like CrewAI and LangGraph, providing a consistent approach to Arcade integration across different agent frameworks

Spartee commented:
Awesome! I know @sdserranog has played with this so I'm assigning him to take a look. From first glance it looks awesome. Maybe break into separate pages?

akuya-ekorot commented:
> Awesome! I know @sdserranog has played with this so I'm assigning him to take a look. From first glance it looks awesome. Maybe break into separate pages?

Broke it into separate pages following existing patterns.

sdserranog commented on pages/home/mastra/use-arcade-tools.mdx:
````diff
@@ -0,0 +1,221 @@
+```typescript
+// Generate a response from the agent
+const response = await githubAgent.generate("List my GitHub repositories");
+console.log(response.content);
````
I think this should be
```suggestion
console.log(response.text);
```

sdserranog replied:
This reverted to .content in the last commit.

sdserranog commented on pages/home/mastra/use-arcade-tools.mdx:
```diff
@@ -0,0 +1,221 @@
+// Or stream the response for a more interactive experience
+const stream = await githubAgent.stream("Create a new issue in my repository");
+
+for await (const chunk of stream) {
+  process.stdout.write(chunk.content);
+}
```
```suggestion
for await (const chunk of stream.textStream) {
  process.stdout.write(chunk);
}
```

sdserranog commented on pages/home/mastra/user-auth-interrupts.mdx:
```diff
@@ -0,0 +1,305 @@
+    console.log("Asking agent to list repositories...");
+    const response = await githubAgent.generate("List my GitHub repositories");
+    console.log("\nAgent response:");
+    console.log(response.content);
```
Same here.

sdserranog commented:
This is fantastic! Thanks for making it happen, @akuya-ekorot!

sdserranog commented on pages/home/mastra/use-arcade-tools.mdx:
````diff
@@ -0,0 +1,221 @@
+
+```typescript
+// Generate a response from the agent
+const response = await githubAgent.generate("List my GitHub repositories");
````
Maybe use a more specific example? Right now, it just answers with something like, "I need to know which organization or user account."

akuya-ekorot commented:
> This is fantastic! Thanks for making it happen, @akuya-ekorot!

Thanks @sdserranog. I've addressed the specific examples and corrected the suggested changes.

sdserranog commented:
@akuya-ekorot, your last commit reverted the changes.

akuya-ekorot commented:
> @akuya-ekorot, your last commit reverted the changes.

Should be fixed
